File: library/src/test/java/com/stealthcopter/networktools/PortScanTest.java
package com.stealthcopter.networktools;

import org.junit.Test;

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
public class PortScanTest {
    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidTimout() throws Exception {
        PortScan.onAddress("127.0.0.1").setTimeOutMillis(-1);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidPortLow() throws Exception {
        PortScan.onAddress("127.0.0.1").setPort(0);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidPortHigh() throws Exception {
        PortScan.onAddress("127.0.0.1").setPort(65536);
    }
}

File: library/src/test/java/com/stealthcopter/networktools/PortScanTest.kt
package com.stealthcopter.networktools

import org.junit.Test

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
class PortScanTest {
    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun testIllegalArgumentThrownOnInvalidTimout() {
        PortScan.onAddress("127.0.0.1").setTimeOutMillis(-1)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun testIllegalArgumentThrownOnInvalidPortLow() {
        PortScan.onAddress("127.0.0.1").setPort(0)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun testIllegalArgumentThrownOnInvalidPortHigh() {
        PortScan.onAddress("127.0.0.1").setPort(65536)
    }
}